sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -332,6 +333,15 @@ public boolean shouldRefreshEndpoints(Utils.ValueHolder<Boolean> canRefreshInBac
         DatabaseAccountLocationsInfo currentLocationInfo = this.locationInfo;
         String mostPreferredLocation = Utils.firstOrDefault(currentLocationInfo.preferredLocations);
 
+        if (StringUtils.isEmpty(mostPreferredLocation)) {
```

Comment:
what if we use the effectivePreferredLocations directly when constructing the locationInfo (around line 62),  then we might do not need to specialize the handling when connectionPolicy.preferredRegions being empty situation